src/olympia/blocklist/admin.py

```diff
@@ -129,21 +138,33 @@ def has_change_permission(self, request, obj=None):
     def add_view(self, request, **kwargs):
         if not self.has_add_permission(request):
             raise PermissionDenied
-        guids_data = request.POST.get('guids', request.GET.get('guids'))
-        context = {}
-        fields = (
-            'input_guids', 'min_version', 'max_version', 'url', 'reason',
-            'include_in_legacy')
+
+        ModelForm = type(
+            self.form.__name__, (self.form,), {
+                'existing_min_version': CharField(widget=HiddenInput),
+                'existing_max_version': CharField(widget=HiddenInput)})
         MultiBlockForm = modelform_factory(
-            MultiBlockSubmit, fields=fields,
+            MultiBlockSubmit, fields=self.add_fields, form=ModelForm,
             widgets={'input_guids': HiddenInput()})
+
+        guids_data = request.POST.get('guids', request.GET.get('guids'))
         if guids_data:
             # If we get a guids param it's a redirect from input_guids_view.
-            form = MultiBlockForm(initial={'input_guids': guids_data})
+            form = MultiBlockForm(initial={
+                'input_guids': guids_data,
+                'existing_min_version': Block.MIN,
+                'existing_max_version': Block.MAX})
         elif request.method == 'POST':
             # Otherwise, if its a POST try to process the form.
             form = MultiBlockForm(request.POST)
-            if form.is_valid():
+            form_valid = form.is_valid()
+            frm_data = form.cleaned_data
+            # Check if the versions specified were the ones we calculated which
+            # Blocks would be updated or skipped on.
+            versions_unchanged = (
+                frm_data['min_version'] == frm_data['existing_min_version'] and
+                frm_data['max_version'] == frm_data['existing_max_version'])
+            if form_valid and versions_unchanged:
```

Comment:
mmm I think you're better off using raw `form.data` - `cleaned_data` isn't necessarily complete when the form is invalid, so this is kinda dangerous to rely on this. Then you don't need to call `form.is_valid()` early.

Reply:
I don't think it makes any difference in this case but 🤷‍♂  https://github.com/mozilla/addons-server/pull/13058/commits/3cd2d2de9137d7a3201274fea8c7bd43d0ae2b46